### ISSUE TITLE
bug in twisted.scripts.twistw

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11403]
|<img alt="syver's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| syver reported|
|-|-|
|Trac ID|trac#389|
|Type|defect|
|Created|2003-11-07 16:02:33Z|

<details><summary>Searchable metadata</summary>

```
trac-id__389 389
type__defect defect
reporter__syver syver
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1068220953000000 1068220953000000
changetime__1469150263949081 1469150263949081
version__ 
owner__ 
cc__syver
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="syver's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|syver commented|
|-|-|

```
#!html
<pre>
Line 55 twistw.py 

The script bails out in an attribute error because application doesn't have a 
processName attribute. I looked at twistd.py and lifted the following line from 
that file which seems to work.
    #app.reportProfile(config['report-profile'], application.processName)
    app.reportProfile(config['report-profile'],
                      service.IProcess(application).processName)
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
/cvs/Twisted/twisted/scripts/twistw.py,v  &lt;--  twistw.py
new revision: 1.3; previous revision: 1.2
Fixed

In general, I urge you to test alpha releases and release candidates
as much as possible -- most of the Twisted team uses various UNIX-based
operating systems, so windows-specific code gets less exercise than
it should.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset be3b2e95bf826a7fd328036821037d78a4efd4b9
```
#!CommitTicketReference repository="" revision="be3b2e95bf826a7fd328036821037d78a4efd4b9"
Merge pull request #389 from twisted/8645-scop-closing-2

Author: scop
Reviewer: markrwilliams, rodrigc
Fixes: #8645

More file/stream closing improvements
```
